wordpress-mobile/WordPress-Android
# Issue/13328 13329 backup restore tracking

Parent #13328 & 13329

This PR adds tracking to the backup download and restore flows.

Removing the DRAFT label - questions have been clarified
(1) Which errors should be tracked ** ONLY Errors that are shown to the user **
- Response errors from post requests
- Other request running - is this really an error? Yes, it is
- Response errors from fetching status
~~Missing/null parameters~~

(2) Request types and escaped strings
- Turning a json object into a string prints escaped quotes - not sure this is the format we want to track
JSON is good to send.

To test:
- Launch App
- Turn on feature flags for backup download, restore, and backup
- Turn on collection information (Me -> App Settings -> Privacy Settings -> Collection information switch)
-----------
- Navigate to the Activity Log
- Tap on more menu for an item
- Tap on the Restore to this point option
- Check log for 🔵 Tracked: jetpack_restore_opened, Properties: {"source":"activity_log"}
- Tap on Restore Site
- Tap on Confirm
- Check log for 🔵 Tracked: jetpack_restore_confirmed, Properties: {"restore_types":"{\"sqls\":true,\"themes\":true,\"contents\":true,\"plugins\":true,\"roots\":true,\"uploads\":true}"}
- Tap on Let me know when finished button
- Check log for 🔵 Tracked: jetpack_restore_notify_me_button_tapped

--------
- Navigate to the Backup
- Tap on more menu for an item
- Tap on the Restore to this point option
- Check log for 🔵 Tracked: jetpack_restore_opened, Properties: {"source":"backup"}
- Close out of restore
------

- Navigate to the Activity Log
- Tap on more menu for an item
- Tap on the Download Backup point option
- Check log for 🔵 Tracked: jetpack_backup_download_opened, Properties: {"source":"activity_log"}
- Tap on Create downloadable file
- Check log for 🔵 Tracked: jetpack_backup_download_confirmed, Properties: {"restore_types":"{\"sqls\":true,\"themes\":true,\"contents\":true,\"plugins\":true,\"roots\":true,\"uploads\":true}"}
- Wait for process to finish
- Tap on Share Link Button
- Check log for 🔵 Tracked: jetpack_backup_download_share_link_tapped
- Tap on Download File Button
- Check log for 🔵 Tracked: jetpack_backup_download_file_download_tapped
- Close out of Backup Download

--------
- Navigate to Backup
- Tap on more menu for an item
- Tap on the Download Backup point option
- Check log for 🔵 Tracked: jetpack_backup_download_opened, Properties: {"source":"backup"}
- Close out of backup download
------

- Navigate to the Activity Log
- Tap on more menu for an item
- Tap on the Download Backup point option
- Tap on Create downloadable file
- Check log for 🔵 Tracked: jetpack_backup_download_notify_me_button_tapped
- Close out of Backup Download

PR submission checklist:
